airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py

```diff
@@ -372,6 +373,9 @@ def request_params(
             params["fields"] = self.convert_array_param(self.fields)
         if stream_slice:
             params.update(stream_slice)
+        if self.include_deleted and self.name in ('ads', 'ad_groups', 'campaigns'):
```

Comment:
I'm not sure that is good idea to filtrate by child names in parent class. Better create some class/mixin with additional functional and add it to child classes.

Reply:
Hi @lazebnyi, what do you think about adding an additional field to the class FullRefreshTiktokStream named include_deleted_params with a default value of {}? This value would be overridden in the corresponding classes, so the code above would look like this:

```
        if self.include_deleted:
            params.update(self.include_deleted_params) # or `include_deleted_request_params`
```

And the corresponding class would look like:

```
class Campaigns(IncrementalTiktokStream):
    include_deleted_params = {"filtering": '{"secondary_status": "CAMPAIGN_STATUS_ALL"}'}
    primary_key = "campaign_id"

    def path(self, *args, **kwargs) -> str:
        return "campaign/get/"
        ```

Would that be acceptable? I find mixins a bit overkill, especially considering that they might lead to method overriding and increase code duplication.

That said, this is only a suggestion. If you prefer mixins, I can certainly update the code to use them.